src/Umbraco.Core/Services/ContentService.cs

```diff
@@ -1131,6 +1131,9 @@ public void MoveToRecycleBin(IContent content, int userId = 0)
                     return;
                 }
 
+                // Add a relation for the item being deleted, so that we can know the original parent for if we need to restore later
+                ApplicationContext.Current.Services.RelationService.Relate(content.Parent(), content, Constants.Conventions.Relations.RelateParentDocumentOnDelete);

```

Comment:
This should be done through event subscription similar to what we do with RelateOnCopy. See this class for inspiration: https://github.com/umbraco/Umbraco-CMS/blob/dev-v7/src/Umbraco.Web/Strategies/RelateOnCopyHandler.cs